redis/client_test.go

```diff
@@ -2538,3 +2655,130 @@ func newInitContextTestSetup(t testing.TB) testSetup {
 		ev:      ev,
 	}
 }
+
+func TestClientTLS(t *testing.T) {
+	t.Parallel()
+
+	ts := newTestSetup(t)
+	rs := RunTSecure(t, nil)
+
+	err := ts.rt.Set("caCert", string(rs.TLSCertificate()))
+	require.NoError(t, err)
+
+	gotScriptErr := ts.ev.Start(func() error {
+		_, err := ts.rt.RunString(fmt.Sprintf(`
+			const redis = new Client({
+				socket: {
+					host: '%s',
+					port: %d,
+					tls: {
+						ca: [caCert],
+					}
+				}
+			});
+
+			redis.sendCommand("PING");
+		`, rs.Addr().IP.String(), rs.Addr().Port))
+
+		return err
+	})
+
+	require.NoError(t, gotScriptErr)
+	assert.Equal(t, 1, rs.HandledCommandsCount())
+	assert.Equal(t, [][]string{
+		{"HELLO", "2"},
+		{"PING"},
+	}, rs.GotCommands())
+}
+
+func TestClientTLSAuth(t *testing.T) {
+	t.Parallel()
+
+	clientCert, clientPKey, err := generateTLSCert()
+	require.NoError(t, err)
+
+	ts := newTestSetup(t)
+	rs := RunTSecure(t, clientCert)
+
+	err = ts.rt.Set("caCert", string(rs.TLSCertificate()))
+	require.NoError(t, err)
+	err = ts.rt.Set("clientCert", string(clientCert))
+	require.NoError(t, err)
+	err = ts.rt.Set("clientPKey", string(clientPKey))
+	require.NoError(t, err)
+
+	gotScriptErr := ts.ev.Start(func() error {
+		_, err := ts.rt.RunString(fmt.Sprintf(`
+			const redis = new Client({
+				socket: {
+					host: '%s',
+					port: %d,
+					tls: {
+						ca: [caCert],
+						cert: clientCert,
+						key: clientPKey
+					}
+				}
+			});
+
+			redis.sendCommand("PING");
+		`, rs.Addr().IP.String(), rs.Addr().Port))
+
+		return err
+	})
+
+	require.NoError(t, gotScriptErr)
+	assert.Equal(t, 1, rs.HandledCommandsCount())
+	assert.Equal(t, [][]string{
+		{"HELLO", "2"},
+		{"PING"},
+	}, rs.GotCommands())
+}
+
+func TestClientTLSRespectsNetworkOPtions(t *testing.T) {
+	t.Parallel()
+
+	clientCert, clientPKey, err := generateTLSCert()
+	require.NoError(t, err)
+
+	ts := newTestSetup(t)
+	rs := RunTSecure(t, clientCert)
+
+	err = ts.rt.Set("caCert", string(rs.TLSCertificate()))
+	require.NoError(t, err)
+	err = ts.rt.Set("clientCert", string(clientCert))
+	require.NoError(t, err)
+	err = ts.rt.Set("clientPKey", string(clientPKey))
+	require.NoError(t, err)
+
+	// Set the redis server's IP to be blacklisted.
+	net, err := lib.ParseCIDR(rs.Addr().IP.String() + "/32")
+	require.NoError(t, err)
+	ts.tb.Dialer.Blacklist = []*lib.IPNet{net}
+
+	gotScriptErr := ts.ev.Start(func() error {
+		_, err := ts.rt.RunString(fmt.Sprintf(`
+			const redis = new Client({
+				socket: {
+					host: '%s',
+					port: %d,
+					tls: {
+						ca: [caCert],
+						cert: clientCert,
+						key: clientPKey
+					}
+				}
+			});
+
+			// This operation triggers a connection to the redis
+			// server under the hood, and should therefore fail, since
+			// the server's IP is blacklisted by k6.
+			redis.sendCommand("PING")
+		`, rs.Addr().IP.String(), rs.Addr().Port))
+
+		return err
+	})
+
+	require.Error(t, gotScriptErr)
+	assert.Equal(t, 0, rs.HandledCommandsCount())
```

Comment:
In general, I prefer to also check what the error says, so it doesn't turn out it is just some general JS syntax error or something like that. 

https://pkg.go.dev/github.com/stretchr/testify/require#ErrorContains can be used for both the check it exists and that it's message contains a given string

Reply:
Good call indeed 👍🏻 Will update the test 🙇🏻